src/axom/quest/examples/quest_distributed_distance_query_example.cpp

```diff
@@ -0,0 +1,881 @@
+  SLIC_INFO(query_str);
+  queryTimer.start();
+  query.computeClosestPoints(query_mesh_node,
+                             query_mesh_wrapper.getCoordsetName());
+  queryTimer.stop();
+
+  // Output some timing stats
+  auto getMinMax =
+    [](double inVal, double& minVal, double& maxVal, double& sumVal) {
+      MPI_Allreduce(&inVal, &minVal, 1, MPI_DOUBLE, MPI_MIN, MPI_COMM_WORLD);
+      MPI_Allreduce(&inVal, &maxVal, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD);
+      MPI_Allreduce(&inVal, &sumVal, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+    };
+  {
+    double minInit, maxInit, sumInit;
+    getMinMax(initTimer.elapsedTimeInSec(), minInit, maxInit, sumInit);
+
+    double minQuery, maxQuery, sumQuery;
+    getMinMax(queryTimer.elapsedTimeInSec(), minQuery, maxQuery, sumQuery);
+
+    SLIC_INFO(axom::fmt::format(
+      "Initialization with policy {} took {{avg:{}, min:{}, max:{}}} seconds",
+      params.policy,
+      sumInit / num_ranks,
+      minInit,
+      maxInit));
+    SLIC_INFO(axom::fmt::format(
+      "Query with policy {} took {{avg:{}, min:{}, max:{}}} seconds",
+      params.policy,
+      sumQuery / num_ranks,
+      minQuery,
+      maxQuery));
+  }
+
+  auto cpPositions =
+    query_mesh_wrapper.getParticleMesh().getNodalVectorField<PointType>(
+      "closest_point");
+
+  if(params.isVerbose())
+  {
+    auto cpIndices =
+      query_mesh_wrapper.getParticleMesh().getNodalScalarField<axom::IndexType>(
+        "cp_index");
+
+    auto cpRank =
+      query_mesh_wrapper.getParticleMesh().getNodalScalarField<axom::IndexType>(
+        "cp_rank");
+
+    SLIC_INFO(axom::fmt::format("Closest points ({}):", cpPositions.size()));
+    for(auto i : IndexSet(cpPositions.size()))
+    {
+      SLIC_INFO(axom::fmt::format("\t{}: {{rank:{}, index:{}, position:{}}}",
+                                  i,
+                                  cpRank[i],
+                                  cpIndices[i],
+                                  cpPositions[i]));
+    }
+  }
+
+  //---------------------------------------------------------------------------
+  // Transform closest points to distances and directions
+  //---------------------------------------------------------------------------
+  using primal::squared_distance;
+
+  auto* distances = query_mesh_wrapper.getDC()->GetField("distance");
+  auto* directions = query_mesh_wrapper.getDC()->GetField("direction");
+
+  {
+    double minPts, maxPts, sumPts;
+    getMinMax(distances->Size(), minPts, maxPts, sumPts);
+    SLIC_INFO(
+      axom::fmt::format(" Query points: {{total:{}, min:{}, max:{}, avg:{}}}",
+                        sumPts,
+                        minPts,
+                        maxPts,
+                        sumPts / num_ranks));
+  }
+
+  mfem::Array<int> dofs;
+  for(auto i : IndexSet(nQueryPts))
+  {
+    const auto& cp = cpPositions[i];
+    (*distances)(i) = sqrt(squared_distance(qPts[i], cp));
```

Comment:
This doesn't override `i`, it just uses `i` as the loop variable.
I uses a `slam::PositionSet` as a personal preference since it has more concise notation.

I changed to variable from `i` to `idx` to (hopefully) make it a bit clearer.

Reply:
oh its the () operator.. got it.. carry on you don't need to change the variable name. i'm just dumb

Reply:
I agree that syntax is a bit confusing since it is a pointer to a `GridFunction`, and it uses `operator()` instead of `operator[]`

Reply:
This is what happens when i look at PRs past my bedtime.